src/java/fr/paris/lutece/plugins/extend/service/extender/history/ResourceExtenderHistoryService.java

```diff
@@ -1,194 +1,205 @@
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ *  1. Redistributions of source code must retain the above copyright notice
+ *     and the following disclaimer.
+ *
+ *  2. Redistributions in binary form must reproduce the above copyright notice
+ *     and the following disclaimer in the documentation and/or other materials
+ *     provided with the distribution.
+ *
+ *  3. Neither the name of 'Mairie de Paris' nor 'Lutece' nor the names of its
+ *     contributors may be used to endorse or promote products derived from
+ *     this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDERS OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ * License 1.0
+ */
+package fr.paris.lutece.plugins.extend.service.extender.history;
+
+import fr.paris.lutece.plugins.extend.business.extender.history.IResourceExtenderHistoryDAO;
+import fr.paris.lutece.plugins.extend.business.extender.history.ResourceExtenderHistory;
+import fr.paris.lutece.plugins.extend.business.extender.history.ResourceExtenderHistoryFilter;
+import fr.paris.lutece.plugins.extend.service.ExtendPlugin;
+import fr.paris.lutece.portal.business.event.ResourceEvent;
+import fr.paris.lutece.portal.service.event.ResourceEventManager;
+import fr.paris.lutece.portal.service.security.LuteceUser;
+import fr.paris.lutece.portal.service.security.SecurityService;
+
+import org.apache.commons.lang3.StringUtils;
+
+import jakarta.transaction.Transactional;
+
+import java.util.List;
+
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+import jakarta.inject.Named;
+import jakarta.enterprise.inject.spi.CDI;
+
+import jakarta.servlet.http.HttpServletRequest;
+
+/**
+ *
+ * ResourceExtenderHistoryService
+ *
+ */
+@ApplicationScoped
+@Named( "extend.resourceExtenderHistoryService" )
+public class ResourceExtenderHistoryService implements IResourceExtenderHistoryService
+{
+    /** The Constant BBEAN_SERVICE. */
+    public static final String BEAN_SERVICE = "extend.resourceExtenderHistoryService";
+
+    private IResourceExtenderHistoryDAO _resourceExtenderHistoryDAO = CDI.current( ).select( IResourceExtenderHistoryDAO.class ).get( );
```

Comment:
Inject the bean 
@Inject 
IResourceExtenderHistoryDAO _resourceExtenderHistoryDAO;